casper/src/main/scala/coop/rchain/casper/rholang/syntax/RuntimeSyntax.scala

```diff
@@ -457,12 +469,12 @@ final class RuntimeOps[F[_]](private val runtime: RhoRuntime[F]) extends AnyVal
     )
 
     // Create return channel as first private name created in deploy term
-    val rand = Tools.unforgeableNameRng(deploy.pk, deploy.data.timestamp)
+    val rand = Blake2b512Random.apply(10)
     import coop.rchain.models.rholang.implicits._
-    val returnName: Par = GPrivate(ByteString.copyFrom(rand.next()))
+    val returnName: Par = GPrivate(ByteString.copyFrom(rand.copy().next()))
```

Comment:
Why copy is needed? The original rand is not used for any other purposes.

Reply:
It is needed. Because `rand.next()` change rand internal state. `rand.next()` is not a pure function

Reply:
Oh, I see now random is shared and previously copy was created here.